models/model.js

```diff
@@ -1,62 +1,93 @@
 const db = require("../db/connection");
-const articles = require("../db/data/test-data/articles");
+
 
 exports.fetchTopics = () => {  
   return db
-    .query(
-        `SELECT * FROM topics`     
-        )
+    .query(`SELECT * FROM topics`)
     .then(({ rows }) => {
-        if (rows.length == 0) {
-            return Promise.reject({status: 404, msg: "This data does not exist"})
-         }
-         else { 
-      return rows;
-         }
+      if (rows.length == 0) {
+        return Promise.reject({status: 404, msg: "This data does not exist"})
+      }
+      else { return rows;}
     })
-    
 };
 
+
+
+exports.fetchUsers = () => {  
+  return db
+    .query(`SELECT * FROM users`)
+    .then(({ rows }) => {
+      if (rows.length == 0) {
+          return Promise.reject({status: 404, msg: "This data does not exist"})
+      }
+      else {return rows}  
+    })
+};
+
+
+
+exports.fetchArticles = () => {  
+  return db
+    .query(`
+    SELECT DISTINCT articles.*, COUNT (comments.comment_id) ::INT AS comment_count 
+    FROM articles
+    JOIN comments ON comments.article_id = articles.article_id
+    GROUP BY articles.article_id
+    ORDER BY articles.created_at DESC`
+    )
+      .then(({ rows }) => {
+        if (rows.length == 0) {
+          return Promise.reject({status: 404, msg: "This article does not exist"})
+        }
+        else {return rows}  
+      }) 
+  };
+
+
+
 exports.fetchArticleByID = (article_id) => {  
-    return db
-    .query( `
+  return db
+    .query(`
     SELECT articles.*, COUNT (comments.comment_id) AS comment_count 
     FROM articles
     JOIN comments ON comments.article_id = articles.article_id
     WHERE articles.article_id = $1
     GROUP BY articles.article_id`, [article_id]
-    
     )
       .then(({ rows }) => {
         if (rows.length == 0) {
-           return Promise.reject({status: 404, msg: "This article does not exist"})
+          return Promise.reject({status: 404, msg: "This article does not exist"})
         }
-        else {
-        return rows[0];
-        }  
-    })
-      
-  };
+        else {return rows[0]}  
+      }) 
+};
 
 
-  exports.fetchUsers = () => {  
-    return db
-      .query(
-          `SELECT * FROM users`     
-          )
+
+exports.fetchCommentsFromArticleByID = (article_id) => {  
+  return db
+    .query(
+    `SELECT comments.comment_id, comments.body, comments.author, comments.created_at, comments.votes
+    FROM comments
+    JOIN articles
+    ON comments.article_id = articles.article_id
+    WHERE comments.article_id = $1
+    ORDER BY created_at DESC`, [article_id]
+    )
+   
       .then(({ rows }) => {
         if (rows.length == 0) {
-            return Promise.reject({status: 404, msg: "This data does not exist"})
-         }
-        else { 
-        return rows;
-        }  
-    })
-      
-  };
+          return Promise.reject({status: 404, msg: "This article does not exist"})
```

Comment:
This is where we will need to check whether an article exists or not. Because if the article does exist, we would still return an empty array for the rows. At the point of discovery that the rows are empty, we need to check for the existence of the article. If it exists, then we can just return the empty rows array and 200. If not, you can return the 404 you already are.